### PR TITLE
Migrated all weave commands to enigma

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

Closes FabricMC/weave#2

The yarn buildscript update can be done at a later time.

Also allow enigma to export tinyv1 mappings under "obf" "deobf" columns.

### CODE REVIEW

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		this.nameDeobf = nameDeobf;
+	}
+
+	@Override
+	public void write(EntryTree<EntryMapping> mappings, MappingDelta<EntryMapping> delta, Path path, ProgressListener progress) {
+		try {
+			Files.deleteIfExists(path);
+			Files.createFile(path);
+		} catch (IOException e) {
+			e.printStackTrace();
```
Comment (gegy1000) [inline src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java]: I believe these should rather throw the exception

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+		node.getChildren().stream()
+				.filter(e -> e instanceof ClassEntry).sorted()
+				.forEach(child -> writeEntry(writer, mappings, child));
+	}
+
+	private void writeClass(Writer writer, ClassEntry entry, Translator translator) {
+		ClassEntry translatedEntry = translator.translate(entry);
+
+		String obfClassName = cuchaz.enigma.utils.Utils.NONE_PREFIX_REMOVER.map(entry.getFullName());
```
Comment (gegy1000) [inline src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java]: Why fully qualified? The class is already imported :P

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV1Helper.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import cuchaz.enigma.translation.representation.MethodDescriptor;
+import cuchaz.enigma.translation.representation.TypeDescriptor;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.Entry;
+import cuchaz.enigma.translation.representation.entry.FieldEntry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+import cuchaz.enigma.utils.Utils;
+
+public class TinyV1Helper {
```
Comment (gegy1000) [inline src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV1Helper.java]: Is this used outside of `TinyMappingsWriter`? Might be worth merging into there

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java (hunk 1 above)
Comment (liach) [reply src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java]: The other writers just print the stacktrace. Might batch change these a bit later.

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV1Helper.java (hunk 3 above)
Comment (liach) [reply src/main/java/cuchaz/enigma/translation/mapping/serde/TinyV1Helper.java]: Done.

File: src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java (hunk 2 above)
Comment (liach) [reply src/main/java/cuchaz/enigma/translation/mapping/serde/TinyMappingsWriter.java]: Fixed.

**Comment by asiekierka** [comment]
FindMappingErrors should IMO become more deeply integrated with Enigma, whereas the Tiny V1 writer needn't be merged - the Tiny V2 one should.

Perhaps, in the light of plugin-ifying Enigma, we should add an interface for registering mapping readers and writers by filetype?

**Comment by liach** [comment]
For plugin api, I'd suggest @gegy1000 finish #135 so that I can work off that.

**Comment by liach** [comment]
Let me reconsider my choices as I added tiny v2 in #146

**Comment by Runemoro** [comment]
TinyfyCommand isn't needed if https://github.com/FabricMC/Enigma/pull/152 is merged (ConvertMappings can convert to any format, including tiny. Also, what's the difference between `FindMappingErrors` and the current `CheckMappingsCommand`?

**Comment by liach** [comment]
Replaced by #152.

**Pull request closed without merging**